### PR TITLE
Move Login button and functionality to search page

### LINKED ISSUE #24: Move login button to search page
The `GithubAuth` component was being used for login up till now. The entire component contains just a navbar and a login button. Since only a Github login is allowed, there is no point having a whole page for login/register. I suggest the following should be done -

Move the `Login with Github` button to the search page at the top-right corner. Whenever someone clicks on that button, start the whole Github auth workflow that was being started through the `GithubAuth` component earlier.

### PR DESCRIPTION
**Pull request by chirag-wadhwa**
Added the login functionality. After receiving the token, the user it redirected to the pathway display screen. Token received is stored in localStorage.

### CODE REVIEW

File: client/src/components/GetPathway/GetPathway.js [hunk 1]
```diff
@@ -0,0 +1,35 @@
...
+                name
+                steps {
+                id
+                name
+                }
+            }
+        }
+    `
+    return (
+        <Query query={GET_PATHWAY}>
```
Comment (jshreyans) [inline client/src/components/GetPathway/GetPathway.js]: @chirag-wadhwa you've used the `Query` component here whereas in the rest of the files, like [GithubAuth.js](https://github.com/megabit-labs/pathways/blob/master/client/src/components/GithubAuth/GithubAuth.js) you make use of GQL hooks (like useMutation). I would recommend that you make a choice between the 2 and stick to that throughout the project. You could either use the `useQuery` hook here or refactor the code in other places and use components there too.

File: client/src/components/SearchPage/Header.js [hunk 2]
```diff
@@ -1,30 +1,84 @@
...
-                                    src="https://ik.imagekit.io/m52sq26n4h/icons8-github-48__1_.png"
-                                />
-                               <text className={classes.text}>Login with GitHub</text> 
-                            </button>
-                        </Nav.Link>
-                    </Nav>
-                </Navbar>
-            </React.Fragment>
+const CLIENT_ID =
+    process.env.REACT_APP_GITHUB_CLIENT_ID || "8b508ba452a263f604b4"
```
Comment (jshreyans) [inline client/src/components/SearchPage/Header.js]: Is this just for testing purposes? I don't see the use of an env variable when you're passing the token anyways.

File: client/src/components/SearchPage/Header.js [hunk 3]
```diff
@@ -1,30 +1,84 @@
...
+            <Navbar className={classes.navbar}>
+                <Nav className={classes.navlinks}>
+                    <Nav.Link href="#">
+                        <img src={Menu} className={classes.apps} />
+                    </Nav.Link>
+                    <Nav.Link href={`https://github.com/login/oauth/authorize?client_id=${CLIENT_ID}&scope=user`}>
+                        <div className={classes.login}>
+                            <img
+                                className={classes.github}
+                                src="https://ik.imagekit.io/m52sq26n4h/icons8-github-48__1_.png"
```
Comment (jshreyans) [inline client/src/components/SearchPage/Header.js]: Why fetch the image everytime? We can store it locally as an `asset` or better yet, use the icon-library. We're using react-ionicons right @dush-t ?

**Review by jshreyans — CHANGES_REQUESTED** [review]
@chirag-wadhwa A couple of general points that need to be taken care of, apart from the changes requested below:
- Remove commented-out code before you commit changes. A couple of stray console logs is okay. But I can see entire code blocks left commented.
- Consider writing a better PR title. I realize that this PR just implements the login functionality, but when you contribute to a large codebase in the future, this will not be received well.

File: client/src/components/SearchPage/Header.js (hunk 2 above)
Comment (dusnat) [reply client/src/components/SearchPage/Header.js]: @jshreyans this is just a way to provide a fallback value when the environment variable is not set. Like a default value. This should be ok.

File: client/src/components/SearchPage/Header.js (hunk 3 above)
Comment (dusnat) [reply client/src/components/SearchPage/Header.js]: Yup, using ionicons

File: client/src/components/SearchPage/Header.js (hunk 2 above)
Comment (jshreyans) [reply client/src/components/SearchPage/Header.js]: Cool then this looks good.

**Review by jshreyans — APPROVED** [review]
Looks good! Nice work 🚀

File: client/src/components/GetPathway/GetPathway.js (hunk 1 above)
Comment (jshreyans) [reply client/src/components/GetPathway/GetPathway.js]: This query wasn't actually needed so it's been removed.

**Pull request merged**